### DISCUSSION TITLE
Issue with Info map on python

### DISCUSSION THREAD

**Discussion by hexie1995** [mapequation/infomap#278] [General]
Hi, I am trying to finish my script on infomap but running into this issue:
I am wondering if I could get some help with this. I understand that the version has been updated from a 2019 post, but I was not able to find corresponding documentation on which attribute each class has. Could you give me a pointer on this and how to fix the issue, please? Thank you so much!

```
---> 36     for node in tree.leafIter():
     37         node_phys_ind.append(node.physIndex)

C:\ProgramData\Anaconda3\lib\site-packages\infomap.py in __getattr__(self, name)
   1429     # Forward to the node it currently points to
   1430     def __getattr__(self, name):
-> 1431         return getattr(self.current(), name)
   1432 
   1433     data = property(

AttributeError: 'InfoNode' object has no attribute 'leafIter()'
```

**Comment by antoneri** [comment]
Hi and thanks for reaching out!

This seems like something from pre-1.0.

Try something like
```python
from infomap import Infomap
im = Infomap()
# read file, add nodes, etc.
im.run()
for node in im.nodes:
    print(node.node_id)
```

Have a look at the current [API docs](https://mapequation.github.io/infomap/python/infomap.html) for more info.
